libsolidity/interface/CompilerStack.cpp

```diff
@@ -1045,6 +1045,16 @@ void CompilerStack::compileContract(
 		solAssert(false, "Assembly exception for deployed bytecode");
 	}
 
+	// Throw an error if EIP-170 limits are exceeded.
+	if (
+		m_evmVersion >= langutil::EVMVersion::spuriousDragon() &&
+		compiledContract.object.bytecode.size() > 0x6000
+	)
+		m_errorReporter.warning(
+			"Contract code size exceeds 24576 bytes (introduced in Spurious Dragon). "
```

Comment:
I think this error message is good enough, but tbh I don't trust my judgment wrt English

Reply:
What about adding something like "Try enabling the compiler optimizations"?

Reply:
... and use a low `runs` value.

Reply:
I was considering crafting the message based on what optimiser or debugging flags were supplied, but that makes testing complicated (e.g. cant use the syntax test but need to use the old testing framework).

What is the exact requirement here?

Reply:
I think the same text could work with any settings if the wording is right. 

Maybe something like "Have you tried enabling the compiler optimization? Are you using a small `runs` value?"

Reply:
Updated.